staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -259,6 +267,52 @@ func (o *ExplainOptions) renderOpenAPIV2(
 	if schema == nil {
 		return fmt.Errorf("couldn't find resource for %q", gvk)
 	}
+	var buf bytes.Buffer
+	newBytes := LimitFieldsDepth(buf.String(), o.Depth)
+	o.Out.Write([]byte(newBytes))
+	return explain.PrintModelDescription(fieldsPath, &buf, schema, gvk, o.Recursive)
+}
+
+// LimitFieldsDepth limits the depth of the fields to the specified maxDepth and returns the updated output
+func LimitFieldsDepth(output string, maxDepth int) string {
```

Comment:
Actually I was expecting changes in the template https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/explain/v2/templates/plaintext.tmpl rather than Go files. Otherwise, that would compete with the template (some rendering happening in template and some in go) which we may not want.